gnome-initial-setup/pages/site/gis-site-page.c

```diff
@@ -0,0 +1,475 @@
+              json_reader_end_member (reader);
+
+              json_reader_read_member (reader, "street");
+              street = json_reader_get_string_value (reader);
+              json_reader_end_member (reader);
+
+              json_reader_read_member (reader, "locality");
+              locality = json_reader_get_string_value (reader);
+              json_reader_end_member (reader);
+
+              json_reader_read_member (reader, "region");
+              region = json_reader_get_string_value (reader);
+              json_reader_end_member (reader);
+
+              json_reader_read_member (reader, "country");
+              country = json_reader_get_string_value (reader);
+              json_reader_end_member (reader);
+
+              site = gis_site_new (id, facility, street, locality, region, country);
+              g_ptr_array_add (sites, site);
+            }
+          else
+            {
+              g_ptr_array_unref (sites);
+              sites = NULL;
+              goto out;
+            }
+          json_reader_end_element (reader);
+        }
+    }
+
+out:
+  g_clear_object (&reader);
+  g_clear_object (&parser);
+
+  return sites;
+}
+
+static void
+gis_site_page_constructed (GObject *object)
+{
+  GisSitePage *page = GIS_SITE_PAGE (object);
+  GisSitePagePrivate *priv = gis_site_page_get_instance_private (page);
+  GError *error;
+  GSettings *settings;
+
+  G_OBJECT_CLASS (gis_site_page_parent_class)->constructed (object);
+
+  error = NULL;
+  set_site (page, NULL);
+
+  // make all the fields insensitive to start
+  manual_check_toggled (GTK_TOGGLE_BUTTON (priv->manual_check), page);
+
+  g_signal_connect (priv->search_entry,
+                    "notify::" GIS_SITE_SEARCH_ENTRY_PROP_SITE,
+                    G_CALLBACK (entry_site_changed), page);
+
+  g_signal_connect (priv->manual_check, "toggled",
+                    G_CALLBACK (manual_check_toggled), page);
+
+  g_signal_connect (priv->id_entry,
+                    "notify::text-length",
+                    G_CALLBACK (update_page_validation), page);
+
+  g_signal_connect (priv->facility_entry,
+                    "notify::text-length",
+                    G_CALLBACK (update_page_validation), page);
+
+  g_signal_connect (priv->street_entry,
+                    "notify::text-length",
+                    G_CALLBACK (update_page_validation), page);
+
+  g_signal_connect (priv->locality_entry,
+                    "notify::text-length",
+                    G_CALLBACK (update_page_validation), page);
+
+  g_signal_connect (priv->region_entry,
+                    "notify::text-length",
+                    G_CALLBACK (update_page_validation), page);
+
+  g_signal_connect (priv->country_entry,
+                    "notify::text-length",
+                    G_CALLBACK (update_page_validation), page);
+
+  gtk_widget_show (GTK_WIDGET (page));
+}
+
+static void
+gis_site_page_dispose (GObject *object)
```

Comment:
This is doing nothing at the moment and could be removed. (However if you intend to keep `priv->current_site` you'll need to free it here.)